rust_snuba/src/processors/generic_metrics.rs

```diff
@@ -316,7 +319,8 @@ impl Parse for CountersRawRow {
             tags_key: tag_keys.iter().map(|k| k.parse::<u64>().unwrap()).collect(),
             tags_indexed_value: vec![0; tag_keys.len()],
             tags_raw_value: tag_values,
-            materialization_version: 2,
+            materialization_version: parse_matview_ver_from_runtime_opt("counter").unwrap_or(2),
```

Comment:
Is there a reason why we're using this `parse_matview...` only on counters, gauges and dists?

Reply:
Because sets are not affected by sampling, so I'm keeping it as is
```rust
materialization_version: 2
```